tests/rptest/services/tls.py

```diff
@@ -173,10 +393,153 @@ def create_cert(self,
         self._exec(f"openssl req -new -config {cfg} "
                    f"-key {key} -out {csr} -batch")
 
-        self._exec(f"openssl ca -config {self.ca.cfg} -keyfile "
-                   f"{self.ca.key} -cert {self.ca.crt} -policy signing_policy "
-                   f"-extensions signing_node_req -in {csr} -out {crt} "
-                   f"-outdir {self._dir.name} -batch")
+        self._exec(
+            f"faketime -f {faketime} openssl ca -config {self.ca.cfg} -keyfile "
+            f"{self.ca.key} -cert {self.ca.crt} -policy signing_policy "
+            f"-extensions signing_node_req -in {csr} -out {crt} "
+            f"-days {self._cert_expiry_days} -outdir {self._dir.name} -batch")
+
+        cert = Certificate(cfg, key, crt, self.ca)
+        self.certs[name] = cert
+        return cert
+
+
+# TODO(oren): Might want to add the ability to generate CRLs, though
```

Comment:
comment: Hmm that's an interesting point.  Technically we wouldn't know if a cert got revoked.  Possibly something to follow up on (though I don't know of any customer asking for it).